### ISSUE TITLE
about cats weights 

### ISSUE THREAD

**Issue by shuttle999** [xavysp/LDC#4]
![image](https://user-images.githubusercontent.com/90813382/201868194-0cd1d732-33dc-49c6-aec0-c233e5b1916a.png)
Hi, I would like to ask about the weights of the cats loss function, as shown in the figure, there are seven sets of data in 'l_weight', is this the weight you use for training all your models (including B3, B4, B5)? Because I copied your code directly without any changes and started training from scratch (default training is B4), but the result is not very satisfactory.
Could you please send me the parameters of the cats weights for training other models? I would be grateful!

**Comment by xavysp** [comment]
Hi, since I used just four blocks I modified a little bit, as you can see:
```
    l_weight = [[0.05, 2.], [0.05, 2.], [0.05, 2.],
                [0.1, 1.], [0.1, 1.], [0.1, 1.],
                [0.01, 4.]]  # for cats loss
```
and they worked properly,  here the cats weights>:

```
        for o in outputs[:3]:
            loss = loss + tracingloss(o, label, tex_factor=0.05, bdr_factor=2., balanced_w=1.1, device=device)
        loss = loss + tracingloss(outputs[3], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
        loss = loss + tracingloss(outputs[4], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
        loss = loss + tracingloss(outputs[5], label, tex_factor=0.01, bdr_factor=4, balanced_w=1.1, device=device)

```
cheers

**Comment by shuttle999** [comment]
> 
OK, Many thanks

**Comment by shuttle999** [comment]
> Hi, below my modificatins:
> 
> ```
>     l_weight = [[0.05, 2.], [0.05, 2.], [0.05, 2.],
>                 [0.1, 1.], [0.1, 1.], [0.1, 1.],
>                 [0.01, 4.]]  # for cats loss
> ```
> 
> whose worked properly, here the cats weights>:
> 
> ```
>         for o in outputs[:3]:
>             loss = loss + tracingloss(o, label, tex_factor=0.05, bdr_factor=2., balanced_w=1.1, device=device)
>         loss = loss + tracingloss(outputs[3], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
>         loss = loss + tracingloss(outputs[4], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
>         loss = loss + tracingloss(outputs[5], label, tex_factor=0.01, bdr_factor=4, balanced_w=1.1, device=device)
> ```
> 
> cheers
So, if I'm using B4 to train, does that mean my code should look like this?
 ```
         for o in outputs[:3]:
             loss = loss + tracingloss(o, label, tex_factor=0.05, bdr_factor=2., balanced_w=1.1, device=device)
         loss = loss + tracingloss(outputs[3], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
         loss = loss + tracingloss(outputs[4], label, tex_factor=0.01, bdr_factor=4, balanced_w=1.1, device=device)
 ```

**Comment by shuttle999** [comment]
>

**Comment by shuttle999** [comment]
> > Hi, below my modificatins:
> > ```
> >     l_weight = [[0.05, 2.], [0.05, 2.], [0.05, 2.],
> >                 [0.1, 1.], [0.1, 1.], [0.1, 1.],
> >                 [0.01, 4.]]  # for cats loss
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > whose worked properly, here the cats weights>:
> > ```
> >         for o in outputs[:3]:
> >             loss = loss + tracingloss(o, label, tex_factor=0.05, bdr_factor=2., balanced_w=1.1, device=device)
> >         loss = loss + tracingloss(outputs[3], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
> >         loss = loss + tracingloss(outputs[4], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
> >         loss = loss + tracingloss(outputs[5], label, tex_factor=0.01, bdr_factor=4, balanced_w=1.1, device=device)
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > cheers
> 
> So, if I'm using B4 to train, does that mean my code should look like this?
> 
> ```
>         for o in outputs[:3]:
>             loss = loss + tracingloss(o, label, tex_factor=0.05, bdr_factor=2., balanced_w=1.1, device=device)
>         loss = loss + tracingloss(outputs[3], label, tex_factor=0.1, bdr_factor=1, balanced_w = 1.1, device=device)
>         loss = loss + tracingloss(outputs[4], label, tex_factor=0.01, bdr_factor=4, balanced_w=1.1, device=device)
> ```
> 
> Because B4 has four side outputs and one fusion output.

**Comment by xavysp** [comment]
Exactly

**Comment by shuttle999** [comment]
> Exactly

OK, Many thanks
